### ISSUE TITLE
Progress Callback FnMut & Write Progress Callback?

### ISSUE THREAD

**Issue by johannesvollmer** [image-rs/image#1488]
I wondered whether it is planned to add progress to the writer eventually, and if the progress callback could be made mutable? I'm sure if a user wanted to actually use the progress callback to update a UI Progressbar, an immutable `Fn` would not work, right? 

## Draft

The `Fn` could maybe even changed to `FnMut` without major refactorings. This change could even be backwards-compatible on first glance, as any `Fn` can also be passed to any function expecting `FnMut`.

**Comment by fintelia** [comment]
I don't think it was a deliberate decision not to make the callback `FnMut`, and we should probably correct the `ImageDecoder` trait accordingly. However, I believe the change might break any downstream crates which implement their own decoders, so it probably belongs in the 0.24 branch.

**Comment by johannesvollmer** [comment]
Yes, implementing a decoder will not necessarily compile when Fn is changed to FnMut, I didn't think of that :)

**Comment by Shnatsel** [comment]
The `read_with_progress` family of methods was removed in 0.25.x, so this is no longer relevant.
